atc/worker/db_worker_provider.go

```diff
@@ -275,3 +279,17 @@ func (provider *dbWorkerProvider) NewGardenWorker(logger lager.Logger, savedWork
 		buildContainersCount,
 	)
 }
+
+func (provider *dbWorkerProvider) GetTeamByID(teamID int) (string) {
```

Comment:
This seems like an odd place to have a function like this. I'd like to avoid turning the `dbWorkerProvider` into a catch-all place for orphaned functions, which is what this feels like it is doing.

Also, since we're making a db call with `provider.dbTeamFactory.GetTeams()` why not make a SQL query that gets us the exact data we need? (`SELECT ... FROM teams WHERE id=teamID`)
Then the `for` loop is not needed to find the team.

Reply:
This is what I did originally, I then thought it was better to use an existing function than add a new one (tried first with `GetByID`)